code/modules/vore/fluffstuff/custom_items_vr.dm

```diff
@@ -710,6 +710,140 @@ obj/item/weapon/material/hatchet/tacknife/combatknife/fluff/katarina/handle_shie
+	implant_type = /obj/item/weapon/implant/reagent_generator/roiz
+
+/mob/living/carbon/human/proc/use_reagent_implant_roiz()
+	set name = "Lay Egg"
+	set desc = "Force Roiz to lay an egg by squeezing into his lower body! This makes the lizard extremely embarrassed, and it looks funny."
+	set category = "Object"
+	set src in view(1)
+
+	//do_reagent_implant(usr)
+	if(!isliving(usr) || !usr.canClick())
+		return
+
+	if(usr.incapacitated() || usr.stat > CONSCIOUS)
+		return
+
+	var/obj/item/weapon/implant/reagent_generator/roiz/rimplant
+	for(var/I in src.contents)
+		if(istype(I, /obj/item/weapon/implant/reagent_generator))
+			rimplant = I
+			break
+	if (rimplant)
+		if(rimplant.reagents.total_volume <= rimplant.transfer_amount)
+			to_chat(src, "<span class='notice'>[pick(rimplant.empty_message)]</span>")
+			return
+
+		var/obj/item/weapon/reagent_containers/food/snacks/egg/roiz/egg = new(get_turf(src))
+
+		var/index = rand(0,3)
+
+		if (usr != src)
+			var/emote = rimplant.emote_descriptor[index]
+			var/verb_desc = rimplant.verb_descriptor[index]
+			var/self_verb_desc = rimplant.self_verb_descriptor[index]
+			usr.visible_message("<span class='notice'>[usr] [verb_desc] [emote]</span>",
+							"<span class='notice'>You [self_verb_desc] [emote]</span>")
+		else
+			src.visible_message("<span class='notice'>[src] [pick(rimplant.short_emote_descriptor)] an egg.</span>",
+								"<span class='notice'>You [pick(rimplant.self_emote_descriptor)] an egg.</span>")
+		if(prob(15))
+			src.visible_message("<span class='notice'>[src] [pick(rimplant.random_emote)].</span>") // M-mlem.
+
+		rimplant.reagents.remove_reagent(rimplant.generated_reagent, rimplant.transfer_amount)
+
+//Egg item
+//-------------
+/obj/item/weapon/reagent_containers/food/snacks/egg/roiz
+	name = "lizard egg"
+	desc = "It's a large lizard egg."
+	icon = 'icons/vore/custom_items_vr.dmi'
+	icon_state = "egg_roiz"
+	filling_color = "#FDFFD1"
+	volume = 12
+	center_of_mass = list("x"=16, "y"=13)
+
+/obj/item/weapon/reagent_containers/food/snacks/egg/roiz/New()
+	..()
+	reagents.add_reagent("egg", 9)
+	bitesize = 4
+
+/obj/item/weapon/reagent_containers/food/snacks/egg/roiz/afterattack(obj/O as obj, mob/user as mob, proximity)
+	if(istype(O,/obj/machinery/microwave))
+		return ..()
+	if(!(proximity && O.is_open_container()))
+		return
+	user << "You crack \the [src] into \the [O]."
+	reagents.trans_to(O, reagents.total_volume)
+	user.drop_from_inventory(src)
+	qdel(src)
+
+/obj/item/weapon/reagent_containers/food/snacks/egg/roiz/throw_impact(atom/hit_atom)
+	..()
+
+/obj/item/weapon/reagent_containers/food/snacks/friedegg/roiz
+	name = "fried lizard egg"
+	desc = "A large, fried lizard egg, with a touch of salt and pepper. It looks rather chewy."
+	icon = 'icons/vore/custom_items_vr.dmi'
+	icon_state = "friedegg"
+	center_of_mass = list("x"=16, "y"=14)
+
+/obj/item/weapon/reagent_containers/food/snacks/friedegg/roiz/New()
+	..()
+	reagents.add_reagent("protein", 9)
```

Comment:
This will also have 12. Is intended? Didn't increase the volume on this one though.